mlonmcu/target/riscv/ara.py

```diff
@@ -0,0 +1,300 @@
+from mlonmcu.target.common import cli, execute
+from mlonmcu.target.metrics import Metrics
+from .riscv import RISCVTarget
+from .util import update_extensions
+
+logger = get_logger()
+
+
+class AraTarget(RISCVTarget):
+    """Target using a Pulpino-like VP running in the GVSOC simulator"""
+
+    FEATURES = RISCVTarget.FEATURES + ["log_instrs"]
+
+    DEFAULTS = {
+        **RISCVTarget.DEFAULTS,
+        "xlen": 64,
+        "abi": "lp64d",
+        "extensions": ["g", "c", "v"],  # TODO overwrite extensions elegantly
+        "nr_lanes": 4,
+        "vlen": 4096,
+    }
+
+    REQUIRED = RISCVTarget.ARA_GCC_TOOLCHAIN_REQUIRED + [
+        "ara.apps_dir",  # for the bsp package usded in compilation
+        "ara.hardware_dir",  # for the rtls
+        "ara.bender_path",  # for simulation
+        "ara.verilator_install_dir",  # for simulation
+        "ara.tb_verilator_build_dir"  # actually just a tmp folder, recommended to be under hardware/build
+    ]
+
+    def __init__(self, name="ara", features=None, config=None):
+        super().__init__(name, features=features, config=config)
+
+    @property
+    def ara_apps_dir(self):
+        return Path(self.config["ara.apps_dir"])
+
+    @property
+    def ara_hardware_dir(self):
+        return Path(self.config["ara.hardware_dir"])
+
+    @property
+    def ara_bender_path(self):
+        return Path(self.config["ara.bender_path"])
+
+    @property
+    def ara_verilator_install_dir(self):
+        return Path(self.config["ara.verilator_install_dir"])
+
+    @property
+    def ara_tb_verilator_build_dir(self):
+        return Path(self.config["ara.tb_verilator_build_dir"])
+
+    @property
+    def Vara_tb_verilator(self):
+        return Path(self.config["ara.tb_verilator_build_dir"]) / "Vara_tb_verilator"
+
+    @property
+    def ara_verilator_exe_path(self):
+        return Path(self.config["ara.verilator_install_dir"]) / "bin" / "verilator"
+
+    @property
+    def abi(self):
+        value = self.config["abi"]
+        return value
+
+    @property
+    def nr_lanes(self):
+        value = self.config["nr_lanes"]
+        return value
+
+    @property
+    def vlen(self):
+        value = self.config["vlen"]
+        return value
+
+    @property
+    def extensions(self):
+        exts = super().extensions
+        return update_extensions(
+            exts,
+            # pext=self.enable_pext,
+            # pext_spec=self.pext_spec,
+            # vext=self.enable_vext,
+            # elen=self.elen,
+            # embedded=self.embedded_vext,
+            # fpu=self.fpu,
+            # variant=self.gcc_variant,
+        )
+
+    def exec(self, program, *args, cwd=os.getcwd(), **kwargs):
+        """Use target to execute an executable with given arguments"""
+
+        # The following is transferred from https://github.com/pulp-platform/ara/blob/main/hardware/Makefile#L165-L167
+        # The purpose is to populate the /build/verilator folder
+        generate_config_args = []
```

Comment:
Please move the verilator related code to a different method and call it from `get_metrics` instead of `exec` which allows to to expose the time for verilating the hardware as extra optional Metric and decouple the verilator output from the actual simulation output.